src/com/amazon/ion/OversizedValueException.java

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazon.ion;
+
+/**
+ * An error thrown when the user requests an IonReader to parse a scalar value that
+ * exceeds the reader's configured maximum buffer size. This is never thrown from
+ * {@link IonReader#next()}; oversized values encountered during next() are
+ * skipped. This may be thrown from any {@link IonReader} `*value()` method,
+ * but only when a maximum buffer size is specified, and only when incremental
+ * reading is disabled. When incremental reading is enabled, all oversized values
+ * will be detected during `IonReader.next()`, as calling that method at the top
+ * level causes the reader to attempt to buffer the entire top-level value. This
+ * custom exception type exists because several IonReader `*value()` interface
+ * methods return primitives, leaving no way to alert the user to failure.
```

Comment:
what does `primitive` vs (what I assume to be) `object` have to do with throwing an exception or not?

Reply:
One way to model the 'oversized' case would be to give `null` that meaning rather than throwing an exception. But methods that return primitives cannot return `null`.

Reply:
Right... Sorry when I read that for some reason I thought "primitive" was referring to something in the Ion datamodel (presumably a scalar) not a _Java_ primitive. Which may be obvious but clearly wasn't too me at the time.